src/EnergyPlus/AirflowNetwork/src/Solver.cpp

```diff
@@ -12128,6 +12191,726 @@ namespace AirflowNetwork {
+                        SumLength += DisSysCompDuctData(TypeNum).L;
+                        MaxRough = max(MaxRough, DisSysCompDuctData(TypeNum).roughness);
+                        DynamicLoss += DisSysCompDuctData(TypeNum).TurDynCoef;
+                        // NodeNum1 = AirflowNetworkLinkageData(AFNLinkNum).NodeNums[0];
+                        DuctSizingSBFlag = true;
+                    }
+                    while (NodeNum1 != NodeSplitter) {
+                        for (AFNLinkNum1 = 1; AFNLinkNum1 <= AirflowNetworkNumOfLinks; AFNLinkNum1++) {
+                            if (NodeNum1 == AirflowNetworkLinkageData(AFNLinkNum1).NodeNums[0]) {
+                                CompNum1 = AirflowNetworkLinkageData(AFNLinkNum1).CompNum;
+                                CompTypeNum1 = AirflowNetworkCompData(CompNum1).CompTypeNum;
+                                if (CompTypeNum1 == iComponentTypeNum::DWC) {
+                                    AirflowNetworkLinkageData(AFNLinkNum1).DuctLineType = iDuctLineType::SupplyTrunk;
+                                    TypeNum = AirflowNetworkCompData(CompNum1).TypeNum;
+                                    SumLength += DisSysCompDuctData(TypeNum).L;
+                                    MaxRough = max(MaxRough, DisSysCompDuctData(TypeNum).roughness);
+                                    DynamicLoss += DisSysCompDuctData(TypeNum).TurDynCoef;
+                                    NodeNum1 = AirflowNetworkLinkageData(AFNLinkNum1).NodeNums[1];
+                                    DuctSizingSTFlag = true;
+                                    break;
+                                }
+                            }
+                        }
+                    }
+                }
+                if (DuctSizingSTFlag) {
+                    Real64 Velocity = 0.0;
+                    Real64 flowrate = DisSysCompCVFData(1).FlowRate / m_state.dataEnvrn->StdRhoAir;
+                    if (AirflowNetworkSimu.ductSizeMethod == AirflowNetworkSimuProp::DuctSizeMethod::MaxVelocity) {
+                        SupplyTrunkD = sqrt(4.0 * flowrate / AirflowNetworkSimu.DuctSizeMaxV / 3.1415926);
+                        SupplyTrunkArea = SupplyTrunkD * SupplyTrunkD / 4.0 * 3.1415926;
+                    } else {
+                        Real64 MaxDiameter = sqrt(4.0 * flowrate / MinVelocity / 3.1415926);
+                        Real64 MinDiameter = sqrt(4.0 * flowrate / MaxVelocity / 3.1415926);
+                        Par(1) = AirflowNetworkSimu.DuctSizePLossSTrunk;
+                        Par(2) = DisSysCompCVFData(1).FlowRate;
+                        Par(3) = SumLength;
+                        Par(4) = DynamicLoss;
+                        Par(5) = MaxRough;
+
+                        General::SolveRoot(m_state, EPS, MaxIte, SolFla, hydraulicDiameter, DuctDResidual, MinDiameter, MaxDiameter, Par);
+                        if (SolFla == -1) {
+                            if (!m_state.dataGlobal->WarmupFlag) {
```

Comment:
At the risk of opening up a can of worms ... `state` is a member here as opposed to being the first argument of the function? Why?

Reply:
Will try to keep the can of worms closed. Including state as an argument in every function means that every function has access to way too much and there's no real change from the global way of doing things. Getting state out of the interface is the first step to getting it out of places where it doesn't really belong. The `Solver` object should have access to the whole shooting match, but objects below that, not so much, and those object shouldn't need to know about state to interact with the solver. I also tend to think it is a more OO way of doing things.

Reply:
Yeah ... I don't know. I'm with you in spirit, but I feel like I've fought and lost this battle already.